java/server/src/org/openqa/selenium/remote/server/handler/mobile/SetNetworkConnection.java

```diff
@@ -35,12 +35,9 @@ public SetNetworkConnection(Session session) {
   @SuppressWarnings("unchecked")
   @Override
   public void setJsonParameters(Map<String, Object> allParameters) throws Exception {
-	  Map<String, Map<String, Object>> parameters = (Map<String, Map<String, Object>>)allParameters.get("parameters");
-	  Map<String, Object> typeMap = parameters.get("type");
-	  
-	  type = new ConnectionType(Boolean.parseBoolean(typeMap.get("wifiEnabled").toString()), 
-			  					Boolean.parseBoolean(typeMap.get("dataEnabled").toString()),
-			  					Boolean.parseBoolean(typeMap.get("airplaneMode").toString()));
+    Map<String, Object> parameters = (Map<String, Object>)allParameters.get("parameters");
+    Long bitmask = (Long) parameters.get("type");
```

Comment:
Don't worry: have found the edit file functionality. Can't see any obvious "I have made the change you suggested" ticky box however